flopy/mf6/utils/codegen/templates/context.py.jinja

```diff
@@ -0,0 +1,25 @@
+# autogenerated file, do not modify
+from os import PathLike, curdir
+from typing import Union
+
+from flopy.mf6.data.mfdatautil import ArrayTemplateGenerator, ListTemplateGenerator
+from flopy.mf6 import mfpackage
+from flopy.mf6 import mfmodel
+{# avoid circular import; some pkgs (e.g. mfnam) are used by mfsimbase.py #}
+{% if base == "MFSimulationBase" %}
```

Comment:
It's true, this would simplify the shim. I considered it but decided on a single root template because I wasn't sure if we'd ultimately need separate templates for package/model/sim in the final refactor. The line between "indispensable concept" and "implementation detail of the current version" is still coming into view, I figure it doesn't matter too much where we draw it now since it will almost certainly move as we develop a better idea what generated classes will look like in flopy4. So I'm happy to change this or keep it as is, whichever people prefer.

Reply:
Starting from https://github.com/wpbonelli/flopy/pull/50 I reworked things. There is now a separate top-level templates for each component type:

- `package.py.jinja`
- `exchange.py.jinja`
- `model.py.jinja`
- `simulation.py.jinja`

The shim now does only a few things too complicated to do in jinja:

- setting class attributes
- naming the parent's `__init__` method parameter
- determining which variables to include an `__init__` method parameter for
- swapping `__init__` method parameters for any referenced subpackages
- miscellaneous `__init__` method logic

I think this turned out a lot cleaner, thanks @deltamarnix for getting it started.